### ISSUE TITLE
pranav: new volunteer via Mozilla Science Lab Collaborate

### ISSUE THREAD

**Issue by pranavkantgaur** [acgetchell/CDT-plusplus#6]
I am working on Constrained Delaunay tetrahedralization which is about approximation of a continuous space with discrete counter-part represented using a complex of tetrahedrons which are constrained by input boundaries. It is heavily reliant on CGAL's Linear cell complex and Delaunay triangulation.

I find your project interesting because as I understand from the description, it adds physical meaning to that tetrahedralization and will introduce me to the study of gravity.<p><br><blockquote>This issue was created by @pranavkantgaur via <a href='http://collaborate.mozillascience.org'>Mozilla Science Lab Collaborate</a></blockquote></p>

**Comment by pranavkantgaur** [comment]
@acgetchell Hi, It seems that I have been added 3 times for this project at [Mozilla science lab](http://www.mozillascience.org/projects/quantumGravity). Can you please suggest how can I remove my other 2 entries?

**Comment by acgetchell** [comment]
Ah, I see that. For website issues I've just contacted them directly, as I cannot edit much of the website beyond the project itself.

sciencelab@mozillafoundation.org

**Comment by pranavkantgaur** [comment]
@acgetchell Can you please suggest from where should I start? I have background in using CGAL.

**Comment by acgetchell** [comment]
Hi Pranav,

Nice to have you on board! Here's a quick overview of the physics behind CDT:

- Spacetime is constructed using a lattices of simplices
- Simplices are n-dimensional triangules
- Curvature of a D-dimensional Delaunay lattice is concentrated on the co-dimension 2 simplices, or “bones"
- The Ricci scalar can be expressed as the deficit angle between the co-dimension 1 faces around the bones, times the Areas of the faces
- The Regge calculus equivalent of the Hilbert action is just the sum over all of the bones,

![regge](https://cloud.githubusercontent.com/assets/434849/6726182/6c5f3358-cdd1-11e4-9091-b862908268aa.png)

- The Einstein-Hilbert action is calculated as a discrete path integral which converges with a cosmological constant,

![path](https://cloud.githubusercontent.com/assets/434849/6726142/d64695c8-cdd0-11e4-9b03-33c28f9de430.png)

- The path integral is summed over all inequivalent triangulations
- There is a well-defined Wick-rotation i->-1 to evaluate path integral
- Specify initial manifold, final manifold, find all ways to fill in space
- Pachner moves + metropolis algorithm pick out inequivalent triangulations
- Transition amplitudes calculable
- Random walk diffuser calculates spectral dimension: d at large scales, 2 at small scales
- Fixing the lengths of the edges to get the simplex area easily and increasing number of simplices is dynamical triangulation
- Enforcing causal structure leads to causal dynamical triangulations
- Pick \alpha as edge length for timelike edges and pick “ergodic” Pachner moves respecting foliation
- What are observables? Hard to extract information

The full details may be found here:

http://arxiv.org/pdf/hep-th/0105267v1.pdf

**Comment by acgetchell** [comment]
Pranav,

I'm currently working on ergodic moves, see:

https://github.com/acgetchell/CDT-plusplus/blob/master/src/S3ErgodicMoves.h 

The relevant tests for (6,2) and (2,6) move functionality (described in the paper I referenced above on Arxiv) is:

https://github.com/acgetchell/CDT-plusplus/blob/master/unittests/S3ErgodicMovesTest.cpp

(Look for the two disabled tests.)

CGAL currently has flip and flip_flippable, which implement the (2,3) and (3,2) moves. Both functions call flip_really behind the scenes, which is on line 2211 of:

https://github.com/CGAL/cgal/blob/1d7da5b79fddc413a3218c0e0b546fa69f15de54/Triangulation_3/include/CGAL/Triangulation_data_structure_3.h

So I'd start there to gain insight on how to do the moves. I've been advised that they result in a no-longer Delaunay triangulation, which is okay for my purposes, but maybe later we should think about rebalancing them.

The (4,4) move is similar, except it doesn't change the numbers of simplices, only the cell handles.

For purposes of this project, all move functionality should be separated into a boolean test to see if the move is possible, and then the actual move part. The logical test to see if the move is possible is necessary in order to implement the Metropolis algorithm, which will be done once the complete set of moves is finished.

**Comment by pranavkantgaur** [comment]
@acgetchell Thanks for pointing out the current state of work. Currently, I am trying to familarize myself with the concept of Casual Dynamical triangulations by getting abstract idea from Wikipedia article, a post in [The physics mill](http://www.thephysicsmill.com/2013/10/13/causal-dynamical-triangulations/) and later by going through the Arxiv paper you suggested.

**Comment by acgetchell** [comment]
Contents of conversation moved to Wiki: https://github.com/acgetchell/CDT-plusplus/wiki
